### PR TITLE
Fix RtlTimeFieldsToTime and RtlTimeToTimeFields implements

### PR DESCRIPTION
**Pull request by RadWolfie**
@ergo720 and I made some extensive tests to RtlTimeFieldsToTime and RtlTimeToTimeFields tests. We discovered there are more faults in Cxbx-Reloaded's kernel implement for these APIs. This pull request resolve the test's failures.

Plus I also clean up the codes to make it look more readable than bunked into single if statement and operations.

Before:
```
Kernel Test Suite
build: 6d76b16
Random seed used is 0
Config File Was Loaded. Only running requested tests.
-----------------------------------------------------
0x0130 - RtlTimeFieldsToTime: Tests Starting
  ERROR(line 112): Expected array time_tests[18].time_expected.QuadPart = 0x0, Got time_tests[18].time_actual.QuadPart = 0x26bffb7ecc000
  ERROR(line 113): Expected array time_tests[18].result_expected = 0x0, Got time_tests[18].result_actual = 0x1
0x0130 - RtlTimeFieldsToTime: One or more tests FAILED
0x0131 - RtlTimeToTimeFields: Tests Starting
  ERROR(line 159): Expected array time_fields_tests[12].expected.Year = 0x28a, Got time_fields_tests[12].actual.Year = 0xffff9416
  ERROR(line 160): Expected array time_fields_tests[12].expected.Month = 0x5, Got time_fields_tests[12].actual.Month = 0xfffffff9
  ERROR(line 161): Expected array time_fields_tests[12].expected.Day = 0xa, Got time_fields_tests[12].actual.Day = 0xfffffff7
  ERROR(line 162): Expected array time_fields_tests[12].expected.Hour = 0x4a6, Got time_fields_tests[12].actual.Hour = 0xfffffffe
  ERROR(line 163): Expected array time_fields_tests[12].expected.Minute = 0xe, Got time_fields_tests[12].actual.Minute = 0xffffffd0
  ERROR(line 164): Expected array time_fields_tests[12].expected.Second = 0x29, Got time_fields_tests[12].actual.Second = 0xfffffffb
  ERROR(line 165): Expected array time_fields_tests[12].expected.Millisecond = 0x333, Got time_fields_tests[12].actual.Millisecond = 0xfffffe23
0x0131 - RtlTimeToTimeFields: One or more tests FAILED
------------------------ End of Tests -----------------------
```

After:
```
Kernel Test Suite
build: 6d76b16
Random seed used is 0
Config File Was Loaded. Only running requested tests.
-----------------------------------------------------
0x0130 - RtlTimeFieldsToTime: Tests Starting
0x0130 - RtlTimeFieldsToTime: All tests PASSED
0x0131 - RtlTimeToTimeFields: Tests Starting
0x0131 - RtlTimeToTimeFields: All tests PASSED
------------------------ End of Tests -----------------------
```

### CODE REVIEW

**Comment by ergo720** [comment]
The PR has removed the logging from the two functions. Are we fine with this?

**Comment by RadWolfie** [comment]
> The PR has removed the logging from the two functions. Are we fine with this?

If others are not fine with it, then I'll need to correct the non-log returns to include the log for RtlTimeFieldsToTime. Honestly, it's really weird some do partial log the return.

My opinion about the log removal is mainly because we have xbox kernel test suite's tests to compare against. Even though the suite's test also include overflow max. Which basically already cover the approval of implementation.

The only thing I did not touch is the year, month, and day. Whatever magic @PatrickvL did, works fine as-is except needed `long long` type for the day format to work correctly. So, I decided to make them all long long for easier to code without need for casting. Heck, even some castings aren't needed either since modulo operator already did that for us. (Hint: msvc didn't complain about that.)

**Comment by PatrickvL** [comment]
Logging has at least two objectives; to allow debugging and troubleshooting API calls (which could indeed be less relevant once we know for sure it's implemented correctly), but logging also functions as a trace of what happened. That second purpose would get lost with this logging removal.
Even so, I don't really mind as logging traces are only important when they're actually needed for some investigation and even then most attention would probably go towards other API 's.

TLDR: removing logging here is not a big deal

**Review by PatrickvL — APPROVED** [review]
Approved without testing.

One final suggestion only slightly related to this work: The magic literal value 1000 could be replaced with a constant MSECSPERSEC or something similar (which might already exist somewhere)

**Comment by RadWolfie** [comment]
> One final suggestion only slightly related to this work: The magic literal value 1000 could be replaced with a constant MSECSPERSEC or something similar (which might already exist somewhere)

It's not listed in Cxbx-Reloaded's source files. Although, I do agree with you to add it.

**Review by ergo720 — APPROVED** [review]
